cranelift/codegen/src/isle_prelude.rs

```diff
@@ -82,7 +82,9 @@ macro_rules! isle_common_prelude_methods {
 
         #[inline]
         fn u64_shl(&mut self, x: u64, y: u64) -> u64 {
-            x << y
+            let y = 0x00000000ffffffff & y;
+            let y = u32::try_from(y).unwrap();
+            x.wrapping_shl(y)
```

Comment:
It seems a little awkward here to do effectively two masks (down to 32 bits for `y`, then to log2(64) == 6 bits in `wrapping_shl`), one hidden and one explicit, and have an unwrap-path that may or may not be branch-folded away. It might be a little clearer to just write `x << (y & 63)`, i.e. do the wrapping ourselves? To my eyes at least, it's much less stuff to parse and understand.

Reply:
(It's also strange and unfortunate that `wrapping_shl` only takes a `u32` amount on a `u64` self; I wonder why...)

Reply:
I believe it takes a u32 on all integer types.